Make "lein run-prod"... actually run.

charles-dyfis-net opened the pull request:
Well, sorta. Depends metasoarous/datview#12 and metasoarous/datsync#16.

charles-dyfis-net committed:
Fix conflict between react and react-with-addons

charles-dyfis-net committed:
Force modern posh, as needed by current datview

charles-dyfis-net committed:
Prevent datomic from overriding guava, which broken advanced compilation

charles-dyfis-net committed:
Move to modern sente, with transit-packer vs flexi-packer

charles-dyfis-net committed:
Use data-frisk 0.3.5 (no more react-with-addons!), reagent 0.6-final

charles-dyfis-net committed:
Fix up dependency tree a little more, deal with figwheel mess

metasoarous commented on project.clj:
```diff
@@ -4,12 +4,15 @@
                  [org.clojure/core.match "0.3.0-alpha4"]
                  [org.clojure/core.typed "0.3.23"]
+                 ;; Force versions to fix prod-mode compilation
+                 [cljsjs/react-with-addons "15.4.0-0"]
```
Can we set it so that `cljsjs/react-with-addons` is required for dev, and `cljsjs/react` for production? My understanding is that the latter ends up being somewhat better optimized. Please correct me if I'm wrong!

charles-dyfis-net replied:
Hmm. I'm not aware of differences in optimization -- seeing it documented neither in the README for [cljsjs/react](https://github.com/cljsjs/packages/tree/master/react) or in [the upstream addons docs](https://facebook.github.io/react/docs/addons.html). If you have a pointer re: that understanding, I wouldn't mind following it up -- that said, AFAIK react-with-addons wasn't used at all until it started being pulled in as a dependency by (older releases of) data-frisk. I made it a dependency at an interim point in this PR, before realizing that we could just upgrade data-frisk and moot the issue.

...actually, per Odinodin/data-frisk-reagent#10, it looks like none of the experimental bits included in the addons package were actually used even in the older release.

BTW, part of my concern here is that React is a big enough piece that if we ended up with different versions (not just with-addons vs base react, but, say, 15.4.0 vs 15.0.0), there could be substantially different behavior between the two modes, to the point that I'd start to worry about whether we could see bugs in production mode that couldn't be reproduced in dev mode.

metasoarous replied:
OK; I must have been mistaken. I think I had that in my mind because the profiling tools are in the addons lib, and those slow things down somewhat, but obviously that would only slow things down if activated. Somehow it had just gotten in my head though... Thanks for clarifying.

metasoarous commented on project.clj:
```diff
@@ -32,7 +32,7 @@
                  [testdouble/clojurescript.csv "0.2.0"]
                  [datascript "0.15.0"]
-                 [posh "0.5.3.3"]
+                 [posh "0.5.4"]
```
Should we do `[posh "0.5.5"]` while we're at it?

metasoarous commented on project.clj:
```diff
@@ -145,9 +146,13 @@
-                     {:client {:source-paths ^:replace ["src/dat/sys/client"]
-                               :compiler {;;:main dat.sys.client.start
-                                          ;;:source-map false
+                     {:client {:source-paths ^:replace ["src/dat/sys/client"] ; no "dev", so no figwheel
```
> no "dev", so no figwheel

Can you clarify what you mean here please?

charles-dyfis-net replied:
Sorry -- it wasn't completely obvious to me what this was doing at first read (using a source path not including `dev`, which is where all the figwheel-dependent bits live). That said, if you think it's a bit too obvious for folks who are already reading context, I'm happy to pull that comment back out.

metasoarous replied:
I'm still a little confused. But I'm not too concerned about it.

charles-dyfis-net committed:
Bump posh to 0.5.5

Contains bugfixes suggested by dat* maintainer; not operating correctly in production mode without.

charles-dyfis-net committed:
Revert pseudo-names in production change

Large increase in output size; need largely mooted w/ source maps.

charles-dyfis-net committed:
Dependency-tree cleanup

Eliminate warnings from lein deps :tree

charles-dyfis-net committed:
Fix figwheel

Specify js-reload only once, in cljsbuild section for dev override

metasoarous reviewed (commented):
Are things ready to merge as far as you're concerned? Fighweel and everything working again? If you want to remove devcards to simplify things, that's fine with me.

charles-dyfis-net committed:
Rev dependencies, fix support for current figwheel-system

charles-dyfis-net committed:
bootstrap: Filter out any `:db/fn` items

A `:db/fn` can have a `:fnref` pointing to a Delay, which Transit refuses to
serialize.

charles-dyfis-net commented:
Okay -- *now* I'm confident we're ready to merge. Apologies about the delay.
